crates/ruma-signatures/src/functions.rs

```diff
@@ -278,11 +278,18 @@ pub fn verify_json(
                 .get(key_id)
                 .ok_or_else(|| Error::new("no key for signature in public_key_map"))?;
 
-            let signature_bytes = decode_config(signature, STANDARD_NO_PAD)?;
+            let verify = |config: Config| {
```

Comment:
This block and the one at line 645 seems to be exactly the same. What do you think to moving the code to a private function?

Reply:
I need to capture the local variables in the closure, is it possible to "wrap" those in a smaller closure, and both calling that private function?

Reply:
I was talking about the whole block. Something like:

```
fn verify_signatures(signature: &str, public_key: &str, object: &CanonicalJsonObject) -> Result<(), Error> {
    let verify = |config: Config| {
        let signature_bytes = decode_config(signature, config)?;
        let public_key_bytes = decode_config(&public_key, config)?;

        verify_json_with(&Ed25519Verifier, &public_key_bytes, &signature_bytes, object)
    };

    #[cfg(feature = "compat")]
    also_try_forgiving_base64(STANDARD_NO_PAD, verify)?;
    #[cfg(not(feature = "compat"))]
    verify(STANDARD_NO_PAD)?;

    Ok(())
}
```

and then calling this from both `verify_event` and `verify_json`.

Reply:
Maybe this is a little bit unrelated to your changes, so I'm fine if you want to keep it as it is now.

Reply:
Hmmm, I agree, but i'm gonna apply rule-of-three here; this only appears two times, so i'll keep it as-is, sound good?